### PR TITLE
support legacy projects filtering

### PR DESCRIPTION
**Pull request by zubenkoivan**

### CODE REVIEW

File: platform_buckets_api/service.py [hunk 1]
```diff
@@ -135,15 +135,20 @@ async def get_bucket(self, id: str) -> BaseBucket:
...
-        )
+        try:
+            return await self._storage.get_bucket_by_name(name, org_name, project_name)
+        except NotExistsError:
+            # project_name could be a username if user is searching for
+            # legacy bucket which does't have project. We need to search
+            # for user bucket also. It is important to check that it is a legacy bucket
+            # before returning it since legacy buckets and project buckets
+            # without org have the same name format.
+            bucket = await self._storage.get_bucket_by_name(name, None, project_name)
```
Comment (YevheniiSemendiak) [inline platform_buckets_api/service.py:148]: So, in this case, we still need to pass `owner=project_name`, don't we?

Comment (zubenkoivan) [reply platform_buckets_api/service.py:148]: No, I removed this parameter, it's not needed. Legacy buckets have project_name = owner, you just need to check that it's a legacy bucket afterwards.

Comment (YevheniiSemendiak) [reply platform_buckets_api/service.py:148]: As I understand, `PROJECT_LABEL` was added only in https://github.com/neuro-inc/platform-buckets-api/pull/398, so buckets created prior to 398's deployment wouldn't have project_name. Or did we add some migration for buckets too, not sure, maybe I'm missing something.

Comment (zubenkoivan) [reply platform_buckets_api/service.py:148]: No, there is no migration. This case is handled and also covered by tests. When there is no project label owner is used as a project_name https://github.com/neuro-inc/platform-buckets-api/blob/master/platform_buckets_api/kube_client.py#L125.
Project label is used only to store project name, it's not used in buckets filtering.

**Review by YevheniiSemendiak — APPROVED** [review]
LGTM!

**Pull request merged**